arrows/serialize/protobuf/convert_protobuf.cxx

```diff
@@ -122,45 +123,96 @@ void convert_protobuf( const kwiver::protobuf::detected_object&  proto_det_objec
   {
     det_object.set_detector_name( proto_det_object.detector_name() );
   }
+
+  // Notes
+  det_object.clear_notes();
+  for ( int i = 0; i < proto_det_object.notes_size(); ++i )
+  {
+    det_object.add_note( proto_det_object.notes(i) );
+  } // end for
+
+  // keypoints
+  det_object.clear_keypoints();
+  auto keypoint_map = proto_det_object.keypoints();
+  for ( const auto & entry : keypoint_map )
+  {
+    ::kwiver::vital::point_2d pt;
+    convert_protobuf( entry.second, pt );
+    det_object.add_keypoint( entry.first, pt );
+  } // end for
+
+  // geo_point
+  ::kwiver::vital::geo_point gpt;
+  convert_protobuf( proto_det_object.geopoint(), gpt );
+  det_object.set_geo_point( gpt );
 }
 
 // ----------------------------------------------------------------------------
-void convert_protobuf( const kwiver::vital::detected_object& det_object,
-                  kwiver::protobuf::detected_object&  proto_det_object )
+void convert_protobuf( const ::kwiver::vital::detected_object& det_object,
+                  ::kwiver::protobuf::detected_object&  proto_det_object )
 {
   proto_det_object.set_confidence( det_object.confidence() );
 
-  kwiver::protobuf::bounding_box *proto_bbox = new kwiver::protobuf::bounding_box();
+  auto* proto_bbox = proto_det_object.mutable_bbox();
   convert_protobuf( det_object.bounding_box(), *proto_bbox );
 
-  proto_det_object.set_allocated_bbox(proto_bbox);  // proto_det_object takes ownership
 
   // We're using type() in "const" (read only) way here.  There's utility
   // in having the source object parameter be const, but type() isn't because
   // its a pointer into the det_object.  Using const_cast here is a middle ground
   // though somewhat ugly
-  if ( const_cast<kwiver::vital::detected_object&>(det_object).type() != NULL )
+  if ( const_cast<::kwiver::vital::detected_object&>(det_object).type() != NULL )
   {
-    kwiver::protobuf::detected_object_type *proto_dot = new kwiver::protobuf::detected_object_type();
-    convert_protobuf( * const_cast<kwiver::vital::detected_object&>(det_object).type(), *proto_dot );
+    auto* proto_dot = proto_det_object.mutable_classifcations();
```

Comment:
Great Job here!
You should not new a protobuf object, call the mutable method, it will create and manage the memory for you!